internal/services/network/subnet_resource.go

```diff
@@ -211,10 +246,47 @@ func resourceSubnetCreate(d *pluginsdk.ResourceData, meta interface{}) error {
 
 	// To enable private endpoints you must disable the network policies for the subnet because
 	// Network policies like network security groups are not supported by private endpoints.
-	privateEndpointNetworkPolicies := d.Get("enforce_private_link_endpoint_network_policies").(bool)
-	privateLinkServiceNetworkPolicies := d.Get("enforce_private_link_service_network_policies").(bool)
-	properties.PrivateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPolicies(expandSubnetPrivateLinkNetworkPolicy(privateEndpointNetworkPolicies))
-	properties.PrivateLinkServiceNetworkPolicies = network.VirtualNetworkPrivateLinkServiceNetworkPolicies(expandSubnetPrivateLinkNetworkPolicy(privateLinkServiceNetworkPolicies))
+	var privateEndpointNetworkPolicies network.VirtualNetworkPrivateEndpointNetworkPolicies
+	var privateLinkServiceNetworkPolicies network.VirtualNetworkPrivateLinkServiceNetworkPolicies
+
+	if features.FourPointOhBeta() {
+		privateEndpointNetworkPoliciesRaw := d.Get("private_endpoint_network_policies_enabled")
+		privateLinkServiceNetworkPoliciesRaw := d.Get("private_link_service_network_policies_enabled")
+
+		privateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPolicies(expandSubnetNetworkPolicy(privateEndpointNetworkPoliciesRaw.(bool)))
+		privateLinkServiceNetworkPolicies = network.VirtualNetworkPrivateLinkServiceNetworkPolicies(expandSubnetNetworkPolicy(privateLinkServiceNetworkPoliciesRaw.(bool)))
+	} else {
+		// Set the legacy default value since they are now computed optional
+		privateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPoliciesEnabled
+		privateLinkServiceNetworkPolicies = network.VirtualNetworkPrivateLinkServiceNetworkPoliciesEnabled
```

Comment:
This is where this one is very strange, with the old attributes false == true(enabled), it is horribly confusing. The service team wants the new defaults to be private endpoint == disabled and privateLinkService == enabled, but to avoid a breaking change I kept the original defaults in the not four point oh beta version to false which is enabled.